numba/core/cgutils.py

```diff
@@ -370,13 +370,20 @@ def alloca_once(builder, ty, size=None, name='', zfill=False):
     """
     if isinstance(size, int):
         size = ir.Constant(intp_t, size)
+    # suspend dwarf emission else it links up python source lines with alloca in
+    # the entry block as well as their actual location and it makes the debug
+    # info "jump about".
+    ref = builder.debug_metadata
+    builder.debug_metadata = None
```

Comment:
I'm assuming at this point that the global compiler lock is going to be held by any caller, or that even if it isn't, the builder won't be the same for distinct callers. Does that seem like a reasonable assumption?

Reply:
I think that's a reasonable assumption.